modules/market/src/main/java/com/opengamma/strata/market/param/ParameterSize.java

```diff
@@ -0,0 +1,345 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.param;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.data.MarketDataName;
+
+/**
+ * The market data name and the associated number of parameters.
+ * <p>
+ * This holds the name and the number of parameters that define it.
+ * For example, the curve name and the number of curve nodes.
+ */
+@BeanDefinition(builderScope = "private")
+public final class ParameterSize
```

Comment:
This is very similar to the object 'CurveParameterSize'. Do we want to keep both?